JsonSyntaxException for getWorkflow

argo version:2.5.2
k8s:V1.13.1
language:java
Class:WorkflowServiceApi,method:getWorkflow
I had trouble using WorkflowServiceApi to get Workflow info,a error has occurred:
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was BEGIN_ARRAY at line 1 column 501 path $.spec.templates[0].steps[0]
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
	... 22 more
Looking forward to your reply soon!

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ VERSION    := HEAD
 # VERSION as GIT_BRANCH must be different
 ifneq ($(VERSION),$(GIT_BRANCH))
 
-SWAGGER    := https://raw.githubusercontent.com/mac9416/argo/fix-parallelsteps-swagger/api/openapi-spec/swagger.json
+SWAGGER    := https://raw.githubusercontent.com/$(GIT_ORG)/argo/$(VERSION)/api/openapi-spec/swagger.json
 
 clients: java
 

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ VERSION    := HEAD
 # VERSION as GIT_BRANCH must be different
 ifneq ($(VERSION),$(GIT_BRANCH))
 
-SWAGGER    := https://raw.githubusercontent.com/$(GIT_ORG)/argo/$(VERSION)/api/openapi-spec/swagger.json
+SWAGGER    := https://raw.githubusercontent.com/mac9416/argo/fix-parallelsteps-swagger/api/openapi-spec/swagger.json
 
 clients: java
 
@@ -36,7 +36,7 @@ dist/java.swagger.json: dist/swagger.json
 java: $(JAVA_CLIENT_JAR)
 
 $(JAVA_CLIENT_JAR): dist/openapi-generator-cli.jar dist/java.swagger.json
-	git submodule update --init java
+	git submodule update --remote --init java
 	cd java && git checkout -b $(GIT_BRANCH) || git checkout $(GIT_BRANCH)
 	rm -Rf java/*
 	java \
@@ -81,6 +81,10 @@ $(JAVA_CLIENT_JAR): dist/openapi-generator-cli.jar dist/java.swagger.json
 	cd java && sed 's/<dependencies>/<dependencies><dependency><groupId>io.kubernetes<\/groupId><artifactId>client-java<\/artifactId><version>5.0.0<\/version><\/dependency>/g' pom.xml > tmp && mv tmp pom.xml
     # I don't like these tests
 	rm -Rf java/src/test
+
+	# Hack to work around Java codegen's apparent lack of support for nested lists.
+	sed -i -e 's/ArrayList<List>/ArrayList<List<WorkflowStep>>/' java/src/main/java/io/argoproj/workflow/models/Template.java
+
 	cd java && mvn package -Dmaven.javadoc.skip
 	cd java && git add .
 	cd java && git diff --exit-code || git commit -m 'Updated to $(JAVA_CLIENT_VERSION)'

